dashboard/app/models/concerns/solution_blocks.rb

```diff
@@ -54,6 +54,7 @@ def add_missing_toolbox_blocks
       # Solution block does not appear in the toolbox, add it
       toolboxified_block = create_toolbox_block block
       if toolbox.xpath('//category').empty?
+        toolbox.root = Nokogiri::XML::Node.new 'xml', toolbox if toolbox.root.nil?
```

Comment:
could you parenthesize this function call? It's not immediately obvious whether this is `new('xml', loolbox) if toolbox.root.nil?` or `new('xml', toolbox if toolbox.root.nil?)`